Extend join operation for events to arbitrary array attributes

matteobachetti opened the pull request.

mgullik reviewed (commented):
Hi @matteobachetti, 
thank you very much for your contribution! 
I listed a few minor comments for your pull request, after you address those, we should be able to approve it. 

Thanks!

matteobachetti commented on stingray/tests/test_events.py:
```diff
@@ -433,3 +300,191 @@ def test_pandas_roundtrip(self):
+
+        ev = EventList()
+        ev_other = EventList(time=[1, 2, 3])
+        ev_new = ev.join(ev_other)
```
When you are creating a new EventList, sometimes you start with an empty one and add data. This is different from adding an empty event list to an existing one. What do you think?

mgullik replied:
I agree, it could be useful to be able to create an empty EventList and fill it with the join routine.

matteobachetti commented on stingray/tests/test_events.py:
```diff
@@ -433,3 +300,191 @@ def test_pandas_roundtrip(self):
+        assert ev_new.pi == None
+        assert ev_new.energy == None
+
+        ev = EventList(time=[1, 2, 3])
```
This one has an empty array as times, the other has no array at all (it gets defined as `None`)

matteobachetti commented:
@mgullik thanks for the review. I think I addressed all your comments but one, where I'm open to changing the behavior but would prefer leaving it as is.

mgullik reviewed (approved):
Hi @matteobachetti, 
thanks for making the requested changes. 

I think we are ready to approve it. I just want to add a bit more of explanation of the Results doc of this routine. Something like this 

I think it could 
        Returns
        -------
        `ev_new` : :class:`EventList` object
            The resulting :class:`EventList` object.
            If the joining EventList (other) is empty a Warning message will appear, and the new EventList will be the same as the starting one. 
            If the joining EventList (other) has the attribute "time" but the other attributes are empty, such as energy and pi, NaNs will be added to the corresponding times of the other EventList. 
            If the starting EventList is empty joining acts as filling the events with the attributes of the other EventList. 

Waht do you think? Do we need more?

matteobachetti commented:
@mgullik I made a slightly different change to the docstring, still along the lines you suggest. Instead of in the return values, I wrote these clarifications in the main text basically.

mgullik commented:
Hi @matteobachetti, 
thanks! I approved your pull request.
